subgraphs/isle-finance-v0/protocols/isle-finance-v0/src/mapping.ts

```diff
@@ -0,0 +1,916 @@
+import {
+  Address,
+  BigDecimal,
+  BigInt,
+  Bytes,
+  ethereum,
+  log,
+} from "@graphprotocol/graph-ts";
+import { ProxyCreated } from "../../../generated/PoolAddressesProvider/PoolAddressesProvider";
+import {
+  Initialized,
+  PoolConfigurator,
+} from "../../../generated/templates/PoolConfigurator/PoolConfigurator";
+import { LoanManager } from "../../../generated/templates/LoanManager/LoanManager";
+import {
+  Pool as PoolTemplate,
+  PoolConfigurator as PoolConfiguratorTemplate,
+  LoanManager as LoanManagerTemplate,
+  WithdrawalManager as WithdrawalManagerTemplate,
+} from "../../../generated/templates";
+import {
+  Deposit,
+  Pool,
+  Withdraw,
+} from "../../../generated/templates/Pool/Pool";
+import {
+  BIGDECIMAL_ZERO,
+  BIGINT_ZERO,
+  exponentToBigDecimal,
+  InterestRateSide,
+  InterestRateType,
+  INT_ZERO,
+  SECONDS_PER_DAY,
+  TokenType,
+  TransactionType,
+} from "../../../src/sdk/constants";
+import { DataManager } from "../../../src/sdk/manager";
+import { TokenManager } from "../../../src/sdk/token";
+import {
+  getProtocolData,
+  ISLE_GLOBALS as ISLE_GLOBALS_ADDRESS,
+  RECEIVABLE as RECEIVABLE_ADDRESS,
+  POOL_ADDRESSES_PROVIDER as POOL_ADDRESSES_PROVIDER_ADDRESS,
+  ISLE_USD_ADDRESS,
+  ISLE_USD_DECIMALS,
+  UNDERLYING_DECIMALS,
+  ZERO_ADDRESS,
+  POOL_CONFIGURATOR_BYTES32 as POOL_CONFIGURATOR_ID,
+  LOAN_MANAGER_BYTES32 as LOAN_MANAGER_ID,
+  WITHDRAWAL_MANAGER_BYTES32 as WITHDRAWAL_MANAGER_ID,
+} from "./constants";
+import { MarketDailySnapshot, Token, _Loan } from "../../../generated/schema";
+import { ERC20, Transfer } from "../../../generated/templates/Pool/ERC20";
+
+/////////////////////
+//// Pool Events ////
+/////////////////////
+
+//
+// Pool-side contracts created event
+export function handleProxyCreated(event: ProxyCreated): void {
+  // create which template depends on the event.params.id
+  const id = event.params.id;
+  if (id === POOL_CONFIGURATOR_ID) {
+    PoolConfiguratorTemplate.create(event.params.proxyAddress);
+  } else if (id === LOAN_MANAGER_ID) {
+    LoanManagerTemplate.create(event.params.proxyAddress);
+  } else if (id === WITHDRAWAL_MANAGER_ID) {
+    WithdrawalManagerTemplate.create(event.params.proxyAddress);
+  }
+}
+
+export function handleInitialized(event: Initialized): void {
+  PoolTemplate.create(event.params.pool_);
+
+  const poolContract = Pool.bind(event.params.pool_);
+  const outputToken = new TokenManager(
+    event.params.pool_,
+    event,
+    TokenType.NON_REBASING
+  );
+
+  const tryInputToken = poolContract.try_asset();
+  if (tryInputToken.reverted) {
+    log.error("[handleInitialized] Pool contract {} does not have an asset", [
+      event.params.pool_.toHexString(),
+    ]);
+    return;
+  }
+
+  const manager = new DataManager(
+    event.params.pool_,
+    tryInputToken.value,
+    event,
+    getProtocolData()
+  );
+  
+  // new market, set both interest rates as 0
+  const market = manager.getMarket();
+  manager.getOrUpdateRate(
```

Comment:
We have fixed interest rates for different loans. Does that make the borrower side become variable or fixed interest?

(no answer myself, but just concerned that this setting might affect future logic)